src/DynamoCoreWpf/Controls/NodeAutoCompleteSearchControl.xaml.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Controls.Primitives;
+using System.Windows.Data;
+using System.Windows.Input;
+using System.Windows.Threading;
+using Dynamo.Utilities;
+using Dynamo.ViewModels;
+using Dynamo.Wpf.ViewModels;
+
+namespace Dynamo.UI.Controls
+{
+    /// <summary>
+    /// Interaction logic for AutoCompleteSearchControl.xaml
+    /// Notice this control shares a lot of logic with InCanvasSearchControl for now
+    /// But they will diverge eventually because of UI improvements to auto complete.
+    /// </summary>
+    public partial class NodeAutoCompleteSearchControl
+    {
+        ListBoxItem HighlightedItem;
+
+        internal event Action<ShowHideFlags> RequestShowNodeAutoCompleteSearch;
```

Comment:
@QilongTang where is this event registered? How does it work?

Reply:
https://github.com/DynamoDS/Dynamo/blob/NodeAutoComplete/src/DynamoCoreWpf/Controls/NodeAutoCompleteSearchControl.xaml.cs#L58.

Reply:
That is the call but what method is it registered with? Which method gets called eventually?

Reply:
I don't think it does anything currently.

Reply:
@aparajit-pratap Sorry it is registered in a different file, see: https://github.com/DynamoDS/Dynamo/blob/NodeAutoComplete/src/DynamoCoreWpf/Views/Core/WorkspaceView.xaml.cs#L110

Reply:
This need to be done on workspace level because we are monitoring all the left click on WS level, so when user left click on anywhere else will dismiss the node autocomplete search bar

Reply:
How is this different from `RequestNodeAutoCompleteSearch` defined on `WorkspaceViewModel`?

Reply:
Sry been so long since I wrote this, I debugged and believe it is leveraged as event driven here : https://github.com/DynamoDS/Dynamo/blob/NodeAutoComplete/src/DynamoCoreWpf/Views/Core/WorkspaceView.xaml#L355

Reply:
My memory faded so my answer may not be clear before.. There are several kinds of show/hide request handled by different events on different level
1. request to show by alt+ left click
2. request to hide by left click on canvas
3. request to hide when window deactivated, e.g. user switched back to Revit, etc

These requests and events was handled on different level for in-canvas search so I inherited the same architecture, hopefully this answers your questions